cpp/harvest/marc_issn_lookup.cc

```diff
@@ -396,22 +436,49 @@ void CleanDuplicationOfField773ByISSN(MARC::Record * const record) {
     }
 }
 
-void CleanDuplicationOfField773ByTitle(MARC::Record * const record) {
-    std::vector<std::string> titles;
-    const ControlNumberGuesser control_number_guesser;
+void CleanDuplicationOfField773ByTitle(MARC::Record * const record, std::vector<TitleInfo773> * const found_title_info_cache) {
     for (auto field(record->begin()); field != record->end(); ++field) {
         if (field->getTag() == "773") {
-            const std::string title(control_number_guesser.NormaliseTitle(field->getFirstSubfieldWithCode('t')));
-            if ((std::find(titles.begin(), titles.end(), title)) != titles.end())
-                record->erase(field);
-            else {
-                titles.emplace_back(title);
-                field->getSubfields().replaceFirstSubfield('t', title);
+            TitleInfo773 ti;
+            ti.Update(*field);
+            if (not ti.subfield_w_.empty()) {
+                if (IsInTitleInfo773Cache(*found_title_info_cache, ti, false))
+                    record->erase(field);
```

Comment:
Can you really erase fields while iterating over them without invalidating the iterator?